ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,515 @@
+    model_data_combined = xr.concat(model_data_list, dim='time')
+    model_data_combined.close()  # Close the files after loading
+
+    # Convert longitudes from -180 to 180 to 0 to 360, if necessary
+    model_longitudes = model_data_combined['longitude'].values
+    if model_longitudes.min() < 0:
+        model_longitudes[model_longitudes < 0] += 360
+        model_data_combined = model_data_combined.assign_coords(longitude=model_longitudes)
+
+    # Convert model time to UNIX time and ensure it's double
+    model_time_seconds = np.array((model_data_combined['time'].values - np.datetime64('1970-01-01T00:00:00')) / np.timedelta64(1, 's')).astype('double')
+    max_model_time = model_time_seconds.max()
+
+    # Defining model_times correctly for use in interpolation
+    model_times = model_time_seconds
+
+    # Load satellite data
+    satellite_data = xr.open_dataset(satellite_file)
+    satellite_longitudes = satellite_data['longitude'].values
+    satellite_longitudes[satellite_longitudes < 0] += 360
+    satellite_times_unix = np.array((satellite_data['time'].values - np.datetime64('1970-01-01T00:00:00')) / np.timedelta64(1, 's')).astype('double')
+    filtered_satellite_times_unix = satellite_times_unix[satellite_times_unix <= max_model_time]
+    filtered_satellite_latitudes = satellite_data['latitude'].values[:len(filtered_satellite_times_unix)]
+    filtered_satellite_longitudes = satellite_longitudes[:len(filtered_satellite_times_unix)]
+
+    # Attempt to read the satellite name from the attributes or a variable
+    if 'sat_name' in satellite_data.variables:
+        satellite_name = satellite_data['sat_name'].values[0]  # Assuming it's stored as a variable
+    elif 'satellite_name' in satellite_data.attrs:
+        satellite_name = satellite_data.attrs['satellite_name']  # Assuming it's stored as an attribute
+    else:
+        satellite_name = "UnknownSatellite"  # Default name if not found
+
+    # Define the Model's Spatial Bounds
+    min_latitude = model_data_combined['latitude'].min()
+    max_latitude = model_data_combined['latitude'].max()
+    min_longitude = model_data_combined['longitude'].min()
+    max_longitude = model_data_combined['longitude'].max()
+
+    # Function to create and apply a spatial mask
+    def spatial_mask(satellite_latitudes, satellite_longitudes):
+        min_lat_val = min_latitude.values.item()
+        max_lat_val = max_latitude.values.item()
+        min_lon_val = min_longitude.values.item()
+        max_lon_val = max_longitude.values.item()
+
+        within_lat_bounds = (satellite_latitudes >= min_lat_val) & (satellite_latitudes <= max_lat_val)
+        within_lon_bounds = (satellite_longitudes >= min_lon_val) & (satellite_longitudes <= max_lon_val)
+        return within_lat_bounds & within_lon_bounds
+
+    def prepare_and_interpolate(model_data, variable_name, satellite_latitudes, satellite_longitudes, satellite_times_unix, valid_times_unix):
```

Comment:
Beyond the difference in the spatial_mask call, are there other reasons that we cannot use the same prepar_and_interpolate as the grib version?

Reply:
@Ghazal-Mohammadpour  can you share more about what you updated?  I see that you still have two functions for  prepare_and_interpolate and I see that spatial_mask has been updated and it seems we have essentially two copies of spatial_mask that are the exact same now.  Is there a reason we still have two prepare_and_interpolate routines and two spatial_mask routines?